aacsbrtables.go

```diff
@@ -239,15 +241,23 @@ func freq_master(data *sbr_extension_data, k0 uint8, k2 uint8, bs_freq_scale uin
 	data.f_master = append(data.f_master, vk1...)
 }
 
-func freq_derived(data *sbr_extension_data, bs_xover_band uint8, k2 uint8) {
+func freq_derived(data *sbr_extension_data, bs_xover_band uint8, k2 uint8) error {
 	data.N_high = data.N_master - bs_xover_band
 	data.N_low = (data.N_high >> 1) + (data.N_high - (data.N_high>>1)<<1)
 
 	data.n = make([]uint8, 2)
 	data.n[0] = data.N_low
 	data.n[1] = data.N_high
 
+	// check for overflow or upper bounds
+	if data.N_high+bs_xover_band+1 < bs_xover_band || len(data.f_master) < int(data.N_high+bs_xover_band+1) {
+		return fmt.Errorf("f_tablehigh invalid index, likely malformed")
+	}
 	data.f_tablehigh = data.f_master[bs_xover_band : data.N_high+bs_xover_band+1]
+	// index out of range [255] with len 0
+	if len(data.f_tablehigh) < int(data.N_high) {
+		return fmt.Errorf("f_tablehigh invalid index, likely malformed")
```

Comment:
Same here, this error could be more informative.

Reply:
Is fix